src/main/drivers/io.c

```diff
@@ -343,7 +343,7 @@ void IOConfigGPIO(IO_t io, ioConfig_t cfg)
 
 void IOConfigGPIOAF(IO_t io, ioConfig_t cfg, uint8_t af)
 {
-    if (!io) {
+    if ((!io) || (IO_GPIOPortIdx(io) < 0) || ((unsigned int)IO_GPIOPortIdx(io) >= ARRAYLEN(ioPortDefs))) {
```

Comment:
The original idea was that you can't get invalid `io` - macros + ioTag mapping tried to ensure that.
It may be reasonable to be defensive here, but invalid `io` error shall be propagated somehow (other functions may fail too and validity check will cause considerable overhead). Also all targets shall implement similar handling.

Another possibility to to use macros / static assert to check that ioPortDefs is defined for all enabled pins / nonzero ioDefUsedMask entries.

Reply:
Please hold fire on this PR. I'm not happy to just be catching a bad param being passed. I'm tracking down where that comes from.